metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py

```diff
@@ -755,6 +778,7 @@ def _sqlglot_lineage_inner(
     )
 
 
+@functools.lru_cache(maxsize=1000)
```

Comment:
Optimization 2: saw lots of duplicate queries, think this is an easy performance gain for what shouldn't be too much memory. Can lower the # if we think SqlParsingResults can be very large, or use a file backed dict